src/test/ui/consts/const-eval/const_fn_ptr_fail2.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(const_fn)]
+#![feature(const_fn_ptr)]
+
+fn double(x: usize) -> usize { x * 2 }
+const X: fn(usize) -> usize = double;
+
+const fn bar(x: fn(usize) -> usize, y: usize) -> usize {
+    x(y)
+}
+
+const Y: usize = bar(X, 2);
```

Comment:
This and Z should not type-check.

Reply:
This is the same issue mentioned in the OP and [here](https://github.com/rust-lang/rust/pull/64635/files/9cf9030e1cc4c6231419f5e9f7a5bd42eb1f55c1#r326856723) and should be fixed by fixing that as well.